src/main/java/org/wildfly/security/util/ServiceLoaderSupplier.java

```diff
@@ -38,31 +41,43 @@
     private final ClassLoader classLoader;
     private int hashCode;
     private volatile E[] result;
+    private final AccessControlContext acc;
 
     public ServiceLoaderSupplier(final Class<E> service, final ClassLoader classLoader) {
         this.service = service;
         this.classLoader = classLoader;
+        this.acc = (System.getSecurityManager() != null) ? AccessController.getContext() : null;
```

Comment:
The context should always be captured here, as the security manager might be initialized *after* one or more configurations are established.